File: src/test/java/kitchenpos/products/tobe/fixtures/FakeProductRepository.java
package kitchenpos.products.tobe.fixtures;

import kitchenpos.products.tobe.Product;
import kitchenpos.products.tobe.ProductRepository;

import java.util.*;

public class FakeProductRepository implements ProductRepository {

    private Map<UUID, Product> inMemory = new HashMap<>();

    @Override
    public Product save(Product product) {
        inMemory.put(product.id(), product);
        return inMemory.get(product.id());
    }

    @Override
    public Optional<Product> findById(UUID id) {
        return Optional.ofNullable(inMemory.get(id));
    }

    @Override
    public List<Product> findAll() {
        return inMemory.values().stream().toList();
    }

    @Override
    public List<Product> findAllByIdIn(List<UUID> ids) {
        return inMemory.values().stream()
                .filter(it -> ids.contains(it.id()))
                .toList();
    }
}

File: src/test/java/kitchenpos/products/tobe/FakeProductRepository.kt
package kitchenpos.products.tobe

import kitchenpos.products.tobe.domain.Product
import kitchenpos.products.tobe.domain.ProductRepository
import java.util.*

class FakeProductRepository : ProductRepository {
    private val productMap: MutableMap<UUID, Product> = mutableMapOf()

    override fun save(product: Product): Product {
        productMap[product.id] = product

        return product
    }

    override fun findById(id: UUID): Product? {
        return productMap[id]
    }

    override fun findAll(): List<Product> {
        return productMap.values.toList()
    }

    override fun findAllByIdIn(ids: List<UUID>): List<Product> {
        return productMap.filterValues { ids.contains(it.id) }.values.toList()
    }
}
